Clean post type drop down

prykon opened the pull request:
Cleaned up the UI for the filter buttons

<img src="https://user-images.githubusercontent.com/36511666/165980732-c48bdee0-6d8a-42f0-a7ca-6d2d2fc5289e.png" width="800px">

corsacca commented:
The UI Looks great!
This part is not longer called, is bulk_edit_on important?
![image](https://user-images.githubusercontent.com/24901539/166419691-ad5f06e5-b032-4a61-9870-54b494b2c031.png)

corsacca commented:
I wonder about making this section more modular in a way that is expandable by plugins
A filter to register the menu items and an action to hook into to output the section.

corsacca commented on archive-template.php:
```diff
@@ -253,19 +253,36 @@ class="second-bar show-for-small-only center list-actions-bar"><!--  /* MOBILE V
+                        <div class="js-sort-dropdown" style="display: inline-block">
+                            <ul class="dropdown menu" data-dropdown-menu>
+                                <li>
+                                    <a href="#">More</a>
```
@prykon  translate.

prykon commented:
> The UI Looks great! This part is not longer called, is bulk_edit_on important? ![image](https://user-images.githubusercontent.com/24901539/166419691-ad5f06e5-b032-4a61-9870-54b494b2c031.png)

If you want I can remove it. Let me know!

prykon commented:
Filter added. Use this snippet to test.

**Add this in a plugin's __construct function:**
`add_filter( 'dt_nav_dropdown_menu_items', [ $this, 'add_menu_items' ] );`

**Add this function to the same plugin:**
`public function add_menu_items( $items ) {
        $items[] = ['key' => 'filter', 'label' => 'Filter', 'post_type' => 'contacts' ];
        return $items;
  }`

prykon commented:
> 

Done! :)

corsacca commented on dt-reports/magic-url-bulk-send.php:
```diff
@@ -14,12 +14,22 @@
-                    <img class="dt-icon" src="<?php echo esc_html( get_template_directory_uri() . '/dt-assets/images/connection.svg' ) ?>"/>
-                </button>
-            </span>
+            <script>
```
@prykon can this section use the `dt_nav_dropdown_menu_items` filter?

prykon replied:
Done.

prykon commented:
Commit a695fd9 and ff7ffc0 details:

- Made dropdown item array element key not be mixed with other array elements
- Removed post_type from dropdown item array. Menu items get filtered by post type when adding the filter, not from the array itself
- Renamed filter for adding items to menu
- Renamed Magic Link Bulk Send Hook to 
- Streamlined Hook that adds the section UI (what displays when you click the menu item) and made it compatible with any plugin, not just the Magic Link Bulk Send
- Icon filename in dropdown item array changed to full icon path
- Check if the dropdown item menu key has a value before adding it

corsacca commented:
Thanks @prykon !
